### PR TITLE
add os and jvm properties to sys.nodes expression

### PR DESCRIPTION
**Pull request by mikethebeer**

### CODE REVIEW

File: sql/src/main/java/io/crate/operation/reference/sys/node/NodeOsInfoExpression.java [hunk 1]
```diff
@@ -43,6 +48,25 @@ public Integer value() {
...
+        });
+        childImplementations.put(OS_ARCH, new OsInfoExpression() {
+            @Override
+            public String value() {
+                return Constants.OS_ARCH;
+            }
+        });
+        childImplementations.put(OS_VERSION, new OsInfoExpression() {
+            @Override
+            public String value() {
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/operation/reference/sys/node/NodeOsInfoExpression.java]: shouldn't the expressions return BytesRef instead of String?

**Comment by mfussenegger** [comment]
aren't there any other tests you have to update?

File: sql/src/main/java/io/crate/operation/reference/sys/node/NodeOsInfoExpression.java (hunk 1 above)
Comment (mikethebeer) [inline sql/src/main/java/io/crate/operation/reference/sys/node/NodeOsInfoExpression.java]: @mfussenegger in es 2.0 already in but with String data type https://github.com/elastic/elasticsearch/blob/master/core/src/main/java/org/elasticsearch/monitor/os/OsProbe.java#L130

File: sql/src/main/java/io/crate/operation/reference/sys/node/NodeOsJvmExpression.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    abstract class JvmExpression extends SysNodeExpression<Object> {
+    }
+
+    public static final String VERSION = "version";
+    public static final String VM_NAME = "vm_name";
+    public static final String VM_VENDOR = "vm_vendor";
+    public static final String VM_VERSION = "vm_version";
+
+    @Inject
```
Comment (kovrus) [inline sql/src/main/java/io/crate/operation/reference/sys/node/NodeOsJvmExpression.java]: what is injected?

**Comment by mfussenegger** [comment]
ack

**Pull request merged**